### PR TITLE
migration: apply backoff only for evacuation migrations

### PR DESCRIPTION
**Pull request by acardace**
Signed-off-by: Antonio Cardace <acardace@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Apply migration backoff only for evacuation migrations.
```

### CODE REVIEW

File: pkg/virt-controller/watch/drain/evacuation/evacuation.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func (c *EvacuationController) addMigration(obj interface{}) {
 	node := ""
 
 	// only observe the migration expectation if our controller created it
-	key, ok := migration.Annotations[virtv1.EvacuationMigrationAnnotation]
+	key, ok := migration.Labels[virtv1.EvacuationMigrationLabel]
```
Comment (rmohr) [inline pkg/virt-controller/watch/drain/evacuation/evacuation.go:183]: You can leave this as an annotation. For the controller it does not really make a difference on which property you filter inside memory.

File: tests/migration_test.go [hunk 2]
```diff
@@ -4467,9 +4466,17 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 
 	})
 
-	Context("when migrating fails", func() {
+	FContext("when evacuating fails", func() {
```
Comment (enp0s3) [inline tests/migration_test.go:4469]: @acardace `FContext` seems like a leftover

**Review by rmohr — COMMENTED** [review]
Let's do the work based on the existing annotation instead of introducing a label. Apart from that it looks like expected :)

File: pkg/virt-controller/watch/drain/evacuation/evacuation.go (hunk 1 above)
Comment (acardace) [reply pkg/virt-controller/watch/drain/evacuation/evacuation.go]: I thought about this but ended up thinking that it'd be nice having a label instead, mainly for looking at things rapidly when using kubectl or to use label selectors in general (if anyone needs that).

But if you are of the opinion that we don't have anyone interested and we shouldn't be introducing a new label I won't be able to bring any justification to the table other than what I just mentioned :)

wdyt?

File: tests/migration_test.go (hunk 2 above)
Comment (acardace) [reply tests/migration_test.go]: ouch! indeed, thanks!

File: pkg/virt-controller/watch/drain/evacuation/evacuation.go (hunk 1 above)
Comment (acardace) [reply pkg/virt-controller/watch/drain/evacuation/evacuation.go]: Ok, I just changed it to use the annotation, I just don't have any arguments for a label :)

**Pull request merged**